src/User.php

```diff
@@ -7013,4 +7013,44 @@ private function cleanInput(array $input): array
 
         return $input;
     }
+
+    /**
+     * User has right for given module and right.
+     *
+     * @param string  $module Module to check
+     * @param integer $right  Right to check
+     *
+     * @return boolean|int
+     **/
+    public function hasRight($module, $right)
+    {
+        /** @var \DBmysql $DB */
+        global $DB;
+        if ($this->isNewItem()) {
+            throw new \LogicException('Cannot check rights for a user not yet saved');
+        } else {
+            $iterator = $DB->request([
+                'SELECT' => 'rights',
+                'FROM'   => 'glpi_profilerights',
+                'JOIN'   => [
+                    'glpi_profiles_users' => [
+                        'ON' => [
+                            'glpi_profilerights'    => 'profiles_id',
+                            'glpi_profiles_users'   => 'profiles_id',
+                        ],
+                    ],
+                ],
+                'WHERE'  => [
+                    'glpi_profilerights.name'       => $module,
+                    'glpi_profilerights.rights'     => ['&', $right],
+                    'glpi_profiles_users.users_id'  => $this->getID(),
+                ],
+            ]);
+
```

Comment:
One last security concern: when the rights for an user are loaded, we do an additional operation by calling `cleanProfile()`:

<img width="552" height="241" alt="Image" src="https://github.com/user-attachments/assets/224998f9-a5c8-4e4b-b917-d90e4509377c" />

Wouldn't something like that be needed here too, to make sure we don't load rights that are out of scope for the profile interface?

Reply:
It's just an SQL SELECT, no profile is loaded, so I don't think it's necessary to reload the profile. Or am I misunderstanding your comment?

Reply:
I'm thinking about the logic from `cleanProfile`, not `reloadCurrentProfile` itself.